python/lsst/ip/isr/linearize.py

```diff
@@ -127,32 +139,280 @@ def fromYaml(self, yamlObject):
+            amp = yamlObject['amplifiers'][ampName]
+            self.linearityCoeffs[ampName] = np.array(amp.get('linearityCoeffs', None), dtype=np.float64)
             self.linearityType[ampName] = amp.get('linearityType', 'None')
             self.linearityBBox[ampName] = amp.get('linearityBBox', None)
 
+        if self.tableData is None:
+            self.tableData = yamlObject.get('tableData', None)
+            if self.tableData:
+                self.tableData = np.array(self.tableData)
+
+        return self
+
     def toDict(self):
         """Return linearity parameters as a dict.
 
         Returns
         -------
         outDict : `dict`:
         """
-        outDict = {'detectorName': self._detectorName,
+        # metadata copied from defects code
+        now = datetime.datetime.utcnow()
+        self.updateMetadata(date=now)
+
+        outDict = {'metadata': self.getMetadata(),
+                   'detectorName': self._detectorName,
                    'detectorSerial': self._detectorSerial,
-                   'hasTable': self._table is not None,
+                   'detectorId': self._detectorId,
+                   'hasTable': self.tableData is not None,
                    'amplifiers': dict()}
         for ampName in self.linearityType:
             outDict['amplifiers'][ampName] = {'linearityType': self.linearityType[ampName],
                                               'linearityCoeffs': self.linearityCoeffs[ampName],
                                               'linearityBBox': self.linearityBBox[ampName]}
+        if self.tableData is not None:
+            outDict['tableData'] = self.tableData.tolist()
+
+        return outDict
+
+    @classmethod
+    def readText(cls, filename):
+        """Read linearity from text file.
+
+        Parameters
+        ----------
+        filename : `str`
+            Name of the file containing the linearity definition.
+        Returns
+        -------
+        linearity : `~lsst.ip.isr.linearize.Linearizer``
+            Linearity parameters.
+        """
+        data = ''
+        try:
+            with open(filename, 'r') as f:
+                data = yaml.load(f, Loader=yaml.FullLoader)
+        except Exception:
+            def yamlLoad(x):
+                return yaml.load(x, Loader=yaml.Loader)
+            with open(filename, 'r') as f:
+                data = yamlLoad(f)
+        return cls().fromYaml(data)
+
+    def writeText(self, filename):
+        """Write the linearity model to a text file.
+
+        Parameters
+        ----------
+        filename : `str`
+            Name of the file to write.
+
+        Returns
+        -------
+        used : `str`
+            The name of the file used to write the data.
+
+        Notes
+        -----
+        The file is written to YAML format and will include any metadata
+        associated with the `Linearity`.
+        """
+        outDict = self.toDict()
+
+        with open(filename, 'w') as f:
```

Comment:
You need to ensure that the file extension is `.yaml` since you can't rely on the caller of `writeText` knowing what the text format will be.